### PR TITLE
top k accuracy

### PR DESCRIPTION
**Pull request by sachinruk**
In early stages of training it's important to see the top-k accuracy to see that your model is indeed training and heading in the right direction. Therefore I've added this as a metric.

```python
import torch
from pytorch_lightning.metrics import TopKAccuracy

target = torch.tensor([0, 1, 2, 3])
preds = torch.tensor([
            [0.0, 0.9, 0.1, 0.0], 
            [0.0, 0.9, 0.1, 0.0], 
            [0.0, 0.9, 0.1, 0.0], 
            [0.0, 0.9, 0.1, 0.0], 
            ])
accuracy = TopKAccuracy(k=2)
accuracy(preds, target) # shows 0.5
```

### CODE REVIEW

**Comment by teddykoker** [comment]
Thanks for the PR? We actually already have an [implementation of top k accuracy](https://github.com/PyTorchLightning/pytorch-lightning/blob/release/1.2-dev/pytorch_lightning/metrics/classification/accuracy.py#L47) in the `release/1.2-dev` branch, which means it will be in the next release!

**Comment by sachinruk** [comment]
oh, that sucks lol. Would you be interested in a weighted accuracy PR? I have an imbalanced dataset, so weights were super important.

**Comment by teddykoker** [comment]
That would be great! We currently have a `sample_weights` argument for [precision_recall](https://github.com/PyTorchLightning/pytorch-lightning/blob/release/1.2-dev/pytorch_lightning/metrics/functional/precision_recall_curve.py#L163), but not for Accuracy. I would recommend adding this functionality to the [`_accuracy_compute`](https://github.com/PyTorchLightning/pytorch-lightning/blob/release/1.2-dev/pytorch_lightning/metrics/functional/accuracy.py#L46) method which is then called by the functional and class accuracy metric.

**Comment by teddykoker** [comment]
Btw, closing this PR, but please reach out to me `teddy` on the [PyTorch Lightning Slack](https://join.slack.com/t/pytorch-lightning/shared_invite/zt-f6bl2l0l-JYMK3tbAgAmGRrlNr00f1A) if you have any questions!
